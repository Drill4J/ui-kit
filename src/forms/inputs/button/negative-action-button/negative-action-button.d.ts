/// <reference types="react" />
interface Props {
    className?: string;
    children: React.ReactNode;
    onClick?: () => void;
    size?: 'large' | 'small';
    disabled?: boolean;
    'data-test'?: string;
}
export declare const NegativeActionButton: ({ className, children, ...rest }: Props) => JSX.Element;
export {};
