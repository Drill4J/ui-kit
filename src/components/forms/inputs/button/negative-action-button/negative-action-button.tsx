import tw, { styled } from 'twin.macro';

import { Button } from '../button';

export const NegativeActionButton = styled(Button)`
  ${tw`border border-red-default bg-red-default text-monochrome-white font-bold
    hover:border-red-medium-tint hover:bg-red-medium-tint
    active:border-red-shade active:bg-red-shade
  `};
`;
