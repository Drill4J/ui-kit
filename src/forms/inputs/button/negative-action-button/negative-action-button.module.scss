@import '../../../../style-guide/index';

.negative-action-button {
  border: 1px solid map-get($red, default);
  background-color: map-get($red, default);
  color: map-get($monochrome, white);
  font-family: $semibold-font;

  &:hover {
    border: 1px solid map-get($red, medium-tint);
    background-color: map-get($red, medium-tint);
  }

  &:active {
    border: 1px solid map-get($red, shade);
    background-color: map-get($red, shade);
  }
}
