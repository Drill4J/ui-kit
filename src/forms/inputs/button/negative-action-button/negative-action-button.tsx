import * as React from 'react';
import { BEM } from '@redneckz/react-bem-helper';

import { Button } from '../button';

import styles from './negative-action-button.module.scss';

interface Props {
  className?: string;
  children: React.ReactNode;
  onClick?: () => void;
  size?: 'large' | 'small';
  disabled?: boolean;
  'data-test'?: string;
}

const negativeActionButton = BEM(styles);

export const NegativeActionButton = negativeActionButton(({ className, children, ...rest }: Props) => (
  <Button className={className} {...rest}>
    {children}
  </Button>
));
