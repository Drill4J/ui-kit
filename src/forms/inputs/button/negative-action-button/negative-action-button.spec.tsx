import React from 'react';
import renderer from 'react-test-renderer';

import { NegativeActionButton } from './negative-action-button';

it('should match snapshot', () => {
  const tree = renderer.create(<NegativeActionButton size="large">Cancel button</NegativeActionButton>).toJSON();

  expect(tree).toMatchSnapshot();
});
