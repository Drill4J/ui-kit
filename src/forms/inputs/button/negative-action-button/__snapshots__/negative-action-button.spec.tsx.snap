// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<button
  className="sc-bdvvaa sc-gsDJrp iJkLDP HIYDZ"
  size="large"
>
  Cancel button
</button>
`;
